src/components/docs-anchor/index.tsx

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable jsx-a11y/anchor-has-content -- We're just passing through props here, so the links should be well formatted */
+import { ProductSlug } from 'types/products'
+import Link from 'next/link'
+
+export default function makeDocsAnchor(
+  product: ProductSlug,
+  basePaths: string[]
+) {
+  const DocsAnchor: React.FC<JSX.IntrinsicElements['a']> = ({
+    href,
+    ...rest
+  }) => {
+    if (!href) return <a {...rest} />
+
+    let adjustedHref = href
+    // TODO: infer this condition dynamically
+    if (basePaths.some((basePath) => href.startsWith(`/${basePath}`))) {
+      // TODO: infer the product dynamically
+      adjustedHref = `/${product}${href}`
+    }
```

Comment:
I think in an ideal world we could take advantage of something like a custom `NavigationContext`, which we've talked in a few other PRs, to get access to `product` and valid base paths.